test/Integration/Examples/Numbers/Numbers.php

```diff
@@ -10,6 +10,6 @@ final class Numbers extends BaseClass
 {
     public function render(): string
     {
-        return (string) (0 + 1234567890 + 42 + 0b10000000000000000000000000000000 + 0b01111111100000000000000000000000 + 0b00000000011111111111111111111111 + 0o755 + 0o644 + 0xFFFFFFFFFFFFFFFFF + 0x123456789ABCDEF + 0xA + 1E3 + 2e6 + 123.456 + 0.1e2 + .22);
+        return (string) (((((((((((((((0 + 1234567890) + 42) + 0b10000000000000000000000000000000) + 0b01111111100000000000000000000000) + 0b00000000011111111111111111111111) + 0o755) + 0o644) + 0xFFFFFFFFFFFFFFFFF) + 0x123456789ABCDEF) + 0xA) + 1E3) + 2e6) + 123.456) + 0.1e2) + .22);
```

Comment:
if you prefer not having those parenthesis here, i experimented with a `shouldAddParenthesisIfNecessary` flag, this way we could transpile without parenthesis (if the operand is the same in chained binary ops)

```php
public function transpile(BinaryOperationNode $binaryOperationNode): string
{
    $expressionTranspiler = new ExpressionTranspiler(
        scope: $this->scope,
        shouldAddQuotesIfNecessary: true,
        shouldAddParenthesisIfNecessary: !(
            $binaryOperationNode->operands->first->root instanceof BinaryOperationNode
            && $binaryOperationNode->operands->first->root->operator === $binaryOperationNode->operator
        )
    );

    $first = $expressionTranspiler->transpile($binaryOperationNode->operands->first);
    $operator = $this->transpileBinaryOperator($binaryOperationNode->operator);
    $second = $expressionTranspiler->transpile($binaryOperationNode->operands->second);

    if ($this->shouldAddParenthesis) {
        return sprintf('(%s %s %s)', $first, $operator, $second);
    } else {
        return sprintf('%s %s %s', $first, $operator, $second);
    }
}
```

Reply:
I don't mind those :) All's good :+1: